spring-kafka/src/main/java/org/springframework/kafka/support/AbstractKafkaHeaderMapper.java

```diff
@@ -97,6 +104,20 @@ public AbstractKafkaHeaderMapper(String... patterns) {
 	 * @param patterns the patterns.
 	 */
 	protected AbstractKafkaHeaderMapper(boolean outbound, String... patterns) {
+		this(outbound, new ArrayList<>(), patterns);
+	}
+
+	/**
+	 * Construct a mapper that will match the supplied patterns (outbound) and all headers
+	 * (inbound). For outbound mapping, certain internal framework headers are never
+	 * mapped. For inbound mapping, Headers that match the pattern specified in
+	 * {@code patternsForListValue} will be appended to the values under the same key.
+	 *
+	 * @param outbound true for an outbound mapper.
+	 * @param patternsForListValue the patterns for multiple values at the same key.
```

Comment:
In terms of Apache Kafka, it is a multi-value headers. Can we follow that style instead ?
I mean the Kafka requirements first and we should map to its concepts even if internally it is a `List` for us. 
so, my suggestion is to rename this property to something like `multiValuePatterns`.
Remember, this is an Open Source? So, code is visible and has to be as clear as possible?

Plus, why bother constructor?
This is an optional feature, so has to be just a setter .
My requirements are to expose via constructor, if we cannot live without that property.
This way the setter for new property could be a varargs even if you set into a `Set` internally .